tests/test_os.py

```diff
@@ -233,12 +233,34 @@ async def test_link():
     """Test the link call."""
     src_filename = join(dirname(__file__), "resources", "test_file1.txt")
     dst_filename = join(dirname(__file__), "resources", "test_file2.txt")
+    initial_src_nlink = stat(src_filename).st_nlink
     await aiofiles.os.link(src_filename, dst_filename)
     assert (
-        exists(src_filename) and
-        exists(dst_filename) and
-        stat(src_filename).st_ino == stat(dst_filename).st_ino
+        exists(src_filename)
+        and exists(dst_filename)
+        and (stat(src_filename).st_ino == stat(dst_filename).st_ino)
+        and (stat(src_filename).st_nlink == initial_src_nlink + 1)(
+            stat(dst_filename).st_nlink == 2
+        )
     )
     await aiofiles.os.remove(dst_filename)
-    assert exists(src_filename) and exists(dst_filename) is False
+    assert (
+        exists(src_filename)
+        and exists(dst_filename) is False
+        and (stat(src_filename).st_nlink == initial_src_nlink)
+    )
```

Comment:
Updated the `aiofiles.os.link` test to better show the difference between symlinks and hardlinks:

- Symlinks: Share the same inode (`st_ino`).
- Hardlinks: Share the same inode (`st_ino`) AND have their link counter (`st_nlink`) incremented.